test/Spec/MockContract.hs

```diff
@@ -376,12 +378,19 @@ mockQueryChainIndex = \case
     throwError @Text "TxFromTxId is unimplemented"
   UtxoSetMembership _ ->
     throwError @Text "UtxoSetMembership is unimplemented"
-  UtxoSetAtAddress _ -> do
+  UtxoSetAtAddress pageQuery _ -> do
     state <- get @MockContractState
     pure $
       UtxoSetAtResponse
         ( state ^. tip
-        , Page (PageSize 100) 1 1 (state ^. utxos ^.. traverse . _1)
+        , pageOf pageQuery (Set.fromList (state ^. utxos ^.. traverse . _1))
+        )
+  UtxoSetWithCurrency pageQuery _ -> do
+    state <- get @MockContractState
+    pure $
+      UtxoSetAtResponse
+        ( state ^. tip
+        , pageOf pageQuery (Set.fromList (state ^. utxos ^.. traverse . _1))
```

Comment:
Is this the full implementation, or is it simply not needed here?

Reply:
The mock implementation for `utxosAt` is a bit oversimplified, simply returns everything in the utxo set.

The actual pagination feature was not implemented fully on the chain-index side at the time I implemented this, but I'll create an issue about it, and fix in a separate PR.